hathor/indexes/addresses_rocksdb_index.py

```diff
@@ -0,0 +1,174 @@
+        assert tx.storage is not None
+        addresses: Set[str] = set()
+
+        def add_address_from_output(output: 'TxOutput') -> None:
+            script_type_out = parse_address_script(output.script)
+            if script_type_out:
+                address = script_type_out.address
+                addresses.add(address)
+
+        for txin in tx.inputs:
+            tx2 = tx.storage.get_transaction(txin.tx_id)
+            txout = tx2.outputs[txin.index]
+            add_address_from_output(txout)
+
+        for txout in tx.outputs:
+            add_address_from_output(txout)
+
+        return addresses
+
+    def publish_tx(self, tx: BaseTransaction, *, addresses: Optional[Iterable[str]] = None) -> None:
+        """ Publish WALLET_ADDRESS_HISTORY for all addresses of a transaction.
+        """
+        if not self.pubsub:
+            return
+        if addresses is None:
+            addresses = self._get_addresses(tx)
+        data = tx.to_json_extended()
+        for address in addresses:
+            self.pubsub.publish(HathorEvents.WALLET_ADDRESS_HISTORY, address=address, history=data)
+
+    def add_tx(self, tx: BaseTransaction) -> None:
+        """ Add tx inputs and outputs to the wallet index (indexed by its addresses).
+        """
+        assert tx.hash is not None
+
+        addresses = self._get_addresses(tx)
+        for address in addresses:
+            self._db.put((self._cf, self._to_key(address, tx)), b'')
+
+        self.publish_tx(tx, addresses=addresses)
+
+    def remove_tx(self, tx: BaseTransaction) -> None:
+        """ Remove tx inputs and outputs from the wallet index (indexed by its addresses).
+        """
+        assert tx.hash is not None
+
+        addresses = self._get_addresses(tx)
+        for address in addresses:
+            self._db.delete((self._cf, self._to_key(address, tx)))
+
+    def handle_tx_event(self, key: HathorEvents, args: 'EventArguments') -> None:
+        """ This method is called when pubsub publishes an event that we subscribed
+        """
+        data = args.__dict__
+        tx = data['tx']
+        meta = tx.get_metadata()
+        if meta.has_voided_by_changed_since_last_call() or meta.has_spent_by_changed_since_last_call():
+            self.publish_tx(tx)
+
+    def _get_from_address_iter(self, address: str) -> Iterable[bytes]:
+        it = self._db.iterkeys(self._cf)
+        it.seek(self._to_key(address))
+        for key in it:
+            addr, _, tx_hash = self._from_key(key)
+            if addr != address:
+                break
+            yield tx_hash
+
+    def get_from_address(self, address: str) -> List[bytes]:
+        """ Get list of transaction hashes of an address
+        """
+        return list(self._get_from_address_iter(address))
+
+    def get_sorted_from_address(self, address: str) -> List[bytes]:
+        """ Get a sorted list of transaction hashes of an address
+        """
+        return list(self._get_from_address_iter(address))
```

Comment:
Should we just use `get_from_address` and explain that `_get_from_address_iter` already returns it sorted?

Reply:
I think it's fine for both to have the same implementation instead of forwarding to `get_from_address`, if that's what you mean.

We seem to have two separate methods because the in-memory implementation needs manual sorting, and the sorted one ends up being a little more expensive and only used on the address_history resource (the rocksdb implementation naturally stores the the indexes sorted, so both methods are identical). I think we can maintain that for now.